ozaria/site/views/play/menu/MapOptionsModal.js

```diff
@@ -0,0 +1,63 @@
+let MapOptionsModal
+const utils = require('core/utils')
+
+const submenuViews = []
+require('app/styles/play/menu/game-menu-modal.sass')
+
+const OptionsView = require('ozaria/site/views/play/menu/OptionsView')
+submenuViews.push(OptionsView)
+
+const ModalView = require('views/core/ModalView')
+const template = require('app/templates/play/menu/game-menu-modal')
+
+require('vendor/scripts/jquery-ui-1.11.1.custom')
+require('vendor/styles/jquery-ui-1.11.1.custom.css')
+
+module.exports = (MapOptionsModal = (function () {
+  MapOptionsModal = class MapOptionsModal extends ModalView {
```

Comment:
why do we need this new modal instead of re-using existing modal?

Reply:
the original modal required `this.level` and `this.session` which is not accessible from the map.

Reply:
you could pass a flag that it's from non-level page and skip those checks and make them return false instead. I already see: `this.options.session?.get('codeLanguage')` , you can do it similarly for level checks to use optional chaining or pass a flag that level is not present and change checks accordingly